google-cloud-spanner/test/google/cloud/spanner/client/transaction_test.rb

```diff
@@ -422,16 +432,78 @@
     mock.expect :create_session, session_grpc, [{ database: database_path(instance_id, database_id), session: nil }, default_options]
     mock.expect :begin_transaction, transaction_grpc, [{ session: session_grpc.name, options: tx_opts}, default_options]
     expect_execute_streaming_sql results_enum, session_grpc.name, "SELECT * FROM users", transaction: tx_selector, seqno: 1, options: default_options
-    mock.expect :commit, commit_resp, [{ session: session_grpc.name, mutations: [], transaction_id: transaction_id, single_use_transaction: nil}, expect_options]
+    mock.expect :commit, commit_resp, [{ session: session_grpc.name, mutations: [], transaction_id: transaction_id, single_use_transaction: nil, return_commit_stats: nil }, expect_options]
     # transaction checkin
     mock.expect :begin_transaction, transaction_grpc, [{ session: session_grpc.name, options: tx_opts}, default_options]
 
     results = nil
-    timestamp = client.transaction call_options: call_options do |tx|
+    resp = client.transaction call_options: call_options do |tx|
       _(tx).must_be_kind_of Google::Cloud::Spanner::Transaction
       results = tx.execute_query "SELECT * FROM users"
     end
-    _(timestamp).must_equal commit_time
+    assert_commit_resp resp
+
+    shutdown_client! client
+
+    mock.verify
+
+    assert_results results
+  end
+
+  it "execute transaction and return commit stats" do
+    mutations = [
+      Google::Cloud::Spanner::V1::Mutation.new(
+        update: Google::Cloud::Spanner::V1::Mutation::Write.new(
+          table: "users", columns: %w(id name active),
+          values: [Google::Cloud::Spanner::Convert.object_to_grpc_value([1, "Charlie", false]).list_value]
+        )
+      )
+    ]
+
+    mock = Minitest::Mock.new
+    mock.expect :create_session, session_grpc, [{ database: database_path(instance_id, database_id), session: nil }, default_options]
+    mock.expect :begin_transaction, transaction_grpc, [{ session: session_grpc.name, options: tx_opts}, default_options]
+    mock.expect :commit, commit_resp_with_stats, [{ session: session_grpc.name, mutations: mutations, transaction_id: transaction_id, single_use_transaction: nil, return_commit_stats: true }, default_options]
+    # transaction checkin
+    mock.expect :begin_transaction, transaction_grpc, [{ session: session_grpc.name, options: tx_opts}, default_options]
+    spanner.service.mocked_service = mock
+
+    resp = client.transaction commit_stats: true do |tx|
+      tx.update "users", [{ id: 1, name: "Charlie", active: false }]
+    end
+    assert_commit_resp resp, stats: true
+
+    shutdown_client! client
+
+    mock.verify
+  end
+
+  it "execute transaction with call options and return commit stats" do
```

Comment:
It's unclear why we add this test. We only need to test the commit stats feature.